### PR TITLE
chore(lint): refactor Markdown linting to use markdownlint-cli2

### PR DESCRIPTION
**Pull request by trentm**
This switches from `markdownlint-cli` to `markdownlint-cli2` and drops
usage of the https://github.com/avto-dev/markdown-lint GitHub action in CI.

- The `avto-dev/markdown-lint` action was using a 4y old version of `markdownlint`.
  AFAICT that action is not being maintained.
- There is a new `npm run lint:markdown` and the `lint.yml` CI workflow uses it.
- The switch from `markdownlint-cli` to `markdownlint-cli2` and `.markdownlint-cli2.jsonc`
  as the config allows using the https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint
  VSCode plugin for in-editor Markdown lint warnings and intellisense in the config file.

Refs: https://github.com/open-telemetry/opentelemetry-js/pull/4713
Refs: #2207

---

The same was done in the core repo in https://github.com/open-telemetry/opentelemetry-js/pull/4713

### CODE REVIEW

**Comment by trentm** [comment]
This updated markdown linting actually catches the errant `### Dependencies` sections described in #2207.

File: .markdownlint-cli2.jsonc [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+{
+  "config": {
+    // https://github.com/DavidAnson/markdownlint/blob/main/README.md#rules--aliases
+    "MD013": false,
+    "MD024": false,
+    "MD033": false,
+    "MD041": false,
+    // XXX Rules to discuss:
+    "MD012": false // no-multiple-blanks, common in CHANGELOG.md files, ~1700 hits in changelog files
+    // "MD004": { "style": "dash" } // ul-style, ~3800 hits in 63 changelog files, XXX discuss just using "consistent" new default for dash
```
Comment (trentm) [inline .markdownlint-cli2.jsonc:11]: @blumamir I'd like to discuss these two rules.

## `"MD012": false`

This is not disabled in the core repo. If re-enabled in the contrib repo here, there are approximately 1700 hits (a) in most of the 63 CHANGELOG.md files and (b) a few hits in these files:

```
   1 plugins/node/opentelemetry-instrumentation-mysql/README.md
   3 .github/ISSUE_TEMPLATE/plugin_request.md
   4 .github/ISSUE_TEMPLATE/feature_request.md
   6 .github/ISSUE_TEMPLATE/bug_report.md
```

I think the cases in the CHANGELOG.md files will just keep happening because they are generated by release-please, we do not want this rule enabled for them. There are two options:

1. globally disable the rule in this config file
2. locally disable the rule for each CHANGELOG.md file. This is slightly burdensome (just a onetime edit) because there are 63 changelog files.

Thoughts?

## `MD004`

Earlier we changed to `style: "dash"` to prefer dashes for bullets. In the core repo we disabled this rule locally in the *2* changelog files. Options:

1. Disable is locally in the 63 changelog files -- similar to how we did in the core repo.
2. Use the newer default of `"consistent"`, which ensures that each file is consistent in the bullets it uses. That means the rule could apply to CHANGELOG.md files as well, but would enforce consistent `*` usage. 
Thoughts?

Comment (blumamir) [reply .markdownlint-cli2.jsonc:11]: > I think the cases in the CHANGELOG.md files will just keep happening because they are generated by release-please, we do not want this rule enabled for them. There are two options:
> 
> 1. globally disable the rule in this config file
> 2. locally disable the rule for each CHANGELOG.md file. This is slightly burdensome (just a onetime edit) because there are 63 changelog files.
> 
> Thoughts?

Another alternative would be to use 2 markdown configs, one for CHANGELOG and one for other markdowns. I don't mind disabling it across the repo, but could be nice to have it if not too complex.



> Earlier we changed to `style: "dash"` to prefer dashes for bullets. In the core repo we disabled this rule locally in the _2_ changelog files. Options:

If we have different config file for CHANGELOGs than this is addressed as well, but if not straight-forward than we can disable it locally in each file or use the `"consistent"` option. What ever is easier and makes more sense to you

Comment (trentm) [reply .markdownlint-cli2.jsonc]: I'd prefer to not have two separate config files, because:
- it would be more likely for those config files to diverge,
- it is a little messy to have yet another config file at the top,
- it will mean markdownlint editor plugins (like the VSCode one referred to above) won't work properly -- the plugin will just use the default `.markdownlint-cli2.jsonc` file

In commit 2e03f596 I've opted to avoid `<!-- markdownlist-disable ... -->` directives in every CHANGELOG.md file by selecting options that work for all .md files. That means `ul-style: consistent` and `no-multiple-blanks: false`.

(A personal preference note: I *like* using multiple blank lines sometimes, both in Markdown and code, to separate bigger sections vs. a single line for related blocks. So I was never a fan of `no-multiple-blank` lines and the eslint equivalent in code.)

File: detectors/node/opentelemetry-resource-detector-alibaba-cloud/CHANGELOG.md [hunk 2]
```diff
@@ -1,77 +1,5 @@
 # Changelog
 
-### Dependencies
```
Comment (blumamir) [inline detectors/node/opentelemetry-resource-detector-alibaba-cloud/CHANGELOG.md:3]: Are these removed manually? will we get more of them with the next releases?

Comment (trentm) [reply detectors/node/opentelemetry-resource-detector-alibaba-cloud/CHANGELOG.md:3]: The removal of these `### Dependencies` sections were all done in this one commit if you want to review them together: https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2234/commits/4733ea4b037d0eb80ed0723414d9648a32423ba8

The issue is described here: https://github.com/open-telemetry/opentelemetry-js-contrib/issues/2207

tl;dr: This *was* a some bug in some old release-please-action that we were using. It isn't happening anymore, so I do **not** expect to get more of these sections.

Removing these sections here is effectively the "A quick fix ..." option described at https://github.com/open-telemetry/opentelemetry-js-contrib/issues/2207

Comment (blumamir) [reply detectors/node/opentelemetry-resource-detector-alibaba-cloud/CHANGELOG.md:3]: 👍

**Review by blumamir — APPROVED** [review]
